src/diagramControl.js

```diff
@@ -273,53 +272,60 @@ class DiagramCtrl extends MetricsPanelCtrl {
+    graphDefinition = this.templateSrv.replaceWithText(graphDefinition);
+    this.diagramType = mermaidAPI.detectType(graphDefinition);
+    var diagramContainer = $(document.getElementById(this.containerDivId));
+    var _this = this;
+    var renderCallback = function(svgCode, bindFunctions) {
+      if (svgCode === '') {
+        diagramContainer.html('There was a problem rendering the graph');
+      } else {
+        diagramContainer.html(svgCode);
+        bindFunctions(diagramContainer[0]);
+        console.debug("Inside rendercallback of renderDiagram");
+        // svgData is empty when this callback happens, update it so the styles will be applied
+        _this.svgData = data;
+        // force a render or we will not see an update
+        _this.render();
+      }
+    };
+    // if parsing the graph definition fails, the error handler will be called but the renderCallback() may also still be called.
+    mermaidAPI.render(this.panel.graphId, graphDefinition, renderCallback);
+  }
+
   updateDiagram(data) {
     if (this.panel.content.length > 0) {
-      this.clearDiagram();
-
       var mode = this.panel.mode;
-      var templatedURL = this.templateSrv.replace(this.panel.mermaidServiceUrl, this.panel.scopedVars);
-
-      function updateDiagram_cont(_this, graphDefinition) {
-        // substitute values inside "link text"
-        // this will look for any composite prefixed with a # and substitute the value of the composite
-        // if a series alias is found, in the form #alias, the value will be substituted
-        // this allows link values to be displayed based on the metric
-        graphDefinition = _this.substituteHashPrefixedNotation(graphDefinition, data);
-        graphDefinition = _this.templateSrv.replaceWithText(graphDefinition);
-        _this.diagramType = mermaidAPI.detectType(graphDefinition);
-        var diagramContainer = $(document.getElementById(_this.containerDivId));
-
-        var renderCallback = function(svgCode, bindFunctions) {
-          if (svgCode === '') {
-            diagramContainer.html('There was a problem rendering the graph');
-          } else {
-            diagramContainer.html(svgCode);
-            bindFunctions(diagramContainer[0]);
-          }
-        };
-        // if parsing the graph definition fails, the error handler will be called but the renderCallback() may also still be called.
-        mermaidAPI.render(_this.panel.graphId, graphDefinition, renderCallback);
-      }
-
       if (mode == 'url') {
+        var templatedURL = this.templateSrv.replace(this.panel.mermaidServiceUrl, this.panel.scopedVars);
         var _this = this;
         this.$http({
           method: 'GET',
-          url: templatedURL
+          url: templatedURL,
+          headers: { 'Accept': 'text/x-mermaid,text/plain;q=0.9,*/*;q=0.8' }
         }).then(function successCallback(response) {
           //the response must have text/plain content-type
```

Comment:
Is this true? We should be accepting `text/x-mermaid` as well since we're asking for it.